heltour/tournament/notify.py

```diff
@@ -36,7 +36,17 @@ def _message_multiple_users(league, usernames, text):
 
 def _lichess_message(league, username, subject, text):
     if league.enable_notifications:
-        lichessapi.send_mail(username, subject, text)
+        try:
+            lichessapi.send_mail(username, subject, text)
+        except lichessapi.ApiClientError as e:
+            reason = str(e)
+            if "The message was rejected" in reason:
+                # If they have blocked us specifically, or if they have
+                # Turned off messages from non-followers In that case,
+                # make an attempt to turn off their lichess noifications.
+                PlayerNotificationSetting.objects.filter(
```

Comment:
Do we need to do this (vs. just failing gracefully)? It's another step for the user to enable lichess messages if they choose to do so.

Reply:
Seems silly to keep trying to send lichess messages to users that are blocking it.  It's not strictly necessary, but it's a wasted request and eats into our request budget for no real gain.  I'm in favour of it, which is why I added it, but if you don't like it, I can remove it.

Reply:
This won't actually work in most cases, because unless the player has made a manual change their PlayerNotificationSetting is [implicit](https://github.com/cyanfish/heltour/blob/3b1ac21e616fbbdbe27164f06cb57b94f7f98dba/heltour/tournament/models.py#L2524). You can try and create a set of objects for the current league, but I'd recommend just removing it since it's a small % of users.

Reply:
Ah, ok.   I'll remove it.

Reply:
Removed.